File: python/refactoring_alphabet/introduceobject.py
class CostCalculator:
    # TODO: introduce an enclosing class for weight and distance
    def calculate_shipping_cost(self,weight, distance) -> float:
        if weight > 10:
            base_cost = 10
        else:
            base_cost = 5
        cost_per_mile = 0.1
        total_cost = base_cost + (distance * cost_per_mile)
        return total_cost 
    
# After start
class ShippingParams:
    def __init__(self, weight, distance):
        self.weight = weight
        self.distance = distance

        
class CostCalculatorAfter:
    def calculate_shipping_cost(params: ShippingParams) -> float:
        if params.weight > 10:
            base_cost = 10
        else:
            base_cost = 5
        cost_per_mile = 0.1
        total_cost = base_cost + (params.distance * cost_per_mile)
        return total_cost        
# After end

File: javascript/refactoring_alphabet/introduceobject.ts
// TODO: introduce an enclosing class for weight and distance

export class CostCalculator {
    calculateShippingCost(weight: number, distance: number): number {
      let baseCost;

      if (weight > 10) {
        baseCost = 10;
      } else {
        baseCost = 5;
      }
      const costPerMile = 0.1;
      const totalCost = baseCost + (distance * costPerMile);
      return totalCost;
    }
  }
  
